### PR TITLE
.then() => .mutate() + .validate()

### PR DESCRIPTION
**Pull request by jeff-mccoy**
- introduce pino for logging json output
- add arch diagram
- add mutate/validate processors 
- add validate examples and test

### CODE REVIEW

File: src/lib/assets/deploy.ts
Comment (cmwylie19) [inline src/lib/assets/deploy.ts:1]: love this code

**Review by cmwylie19 — DISMISSED** [review]
lgtm, i look forward to getting more and more familiar with this code base and would love to help out on the e2e tests

**Review by rjferguson21 — DISMISSED** [review]

**Review by bdw617 — DISMISSED** [review]
smoke tested the changes, looks good. great path

**Review by bdw617 — APPROVED** [review]
the delta looks ok, I reviewed it

**Review by rjferguson21 — APPROVED** [review]

**Pull request merged**